spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaServerGracefulShutdownLifecycle.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.spring;
+
+import org.springframework.context.SmartLifecycle;
+
+import com.linecorp.armeria.server.Server;
+
+/**
+ * Make Armeria {@link Server} utilize spring's SmartLifecycle feature.
+ * So Armeria will shutdown before other web servers and beans in the context.
+ */
+final class ArmeriaServerGracefulShutdownLifecycle implements SmartLifecycle {
+    /**
+     * {@link Server} created by {@link ArmeriaAutoConfiguration}. .
+     */
```

Comment:
Let's remove all Javadocs for the field and methods because it's a package-private class now. 😄

Reply:
I find the class Javadoc, `stop()`, `start()`, `getPhase()` and `isAutoStartup()` useful, so maybe better removing only what's already specified in `SmartLifecycle`.